_posts/2017-03-10-dataflow-cloud-function-trigger.md

```diff
@@ -0,0 +1,83 @@
+
+- The `pipeline-example-trigger` function has to be exported to act as the entry point used by GCF.
+- The incoming `raw-event` that the function receives when a something happens in the bucket gets parsed and the `name` and the `bucket` fields are extracted.
+- The `execute-jar-file` function uses `node-jre` to spawn a java process that starts the `pipeline-standalone.jar` executable with the necessary arguments.
+- The `-> proc` parts are there to handle logging and events from the child process and making sure the
+Cloud Function callback gets called when the java process is done submitting the Dataflow pipeline.
+
+## Deploying the function to GCF
+Now we're ready to deploy the function to see it in action. I use the [Google Cloud CLI](https://cloud.google.com/sdk/) to deploy and start the function on GCF, like this:
+<pre class="highlight">gcloud beta functions deploy pipeline-example-trigger \
+    --local-path target/pipeline-example \
+    --stage-bucket [STAGE BUCKET] \
+    --trigger-bucket [BUCKET TO WATCH]
+</pre>
+
+The `--local-path and --stage-bucket` arguments are the locations the source of the functions should be copied from and to. `--trigger-bucket` is the GCS bucket this function should be watching for changes.
+
+## The proof is in the pudding
+
+![Running function]({{ site.url }}/images/cloudfunctions/pipeline-example-trigger.png)
+
+The function is up and running, and as you can see on the right of the graph there it received an event when I used the CLI to simulate a file change in the bucket it is watching.
+<pre class="highlight">gcloud beta functions call pipeline-example-trigger \
+--data '{"name": "file-name", "bucket": "bucket-name"}'
+</pre>
+
+And the logging works as expected.
+<p> </p>
+![Log]({{ site.url }}/images/cloudfunctions/log.png)
+
+## Closing thoughts
+
+Google Cloud Functions had just entered beta state when this post was written, so understandably it has some rough edges and some missing features. E.g. I'd like a way of providing my own zip file to deploy instead of having to point to a directory with sources because it'd make using a CI/CD service like CircleCI a lot easier. Being able to write functions in a JVM language instead of just Javascript would also make the exercise I've done here, albeit fun and interesting, unnecessary.
+
+The turnaround for deploying a function to test it is a bit long, especially when you include a quite large jar file like here. Luckily Google has created an [emulator](https://github.com/GoogleCloudPlatform/cloud-functions-emulator) that allows us to test the functions locally before deploying them to the cloud. The fact that I developed this in Clojurescript, that has a great REPL for interactive development, also limited the need for actually deploying the code to test it.
+
+My prediction is that we here at Unacast are going to use Google Cloud Functions to simplify and get rid of a lot of complex code in the months to come. Exciting times ahead!
```

Comment:
Maybe add a "Did you like this post? We have plenty more" and reference some of the other posts related or not?